### PR TITLE
feat: add cilium cni instalation

### PR DESCRIPTION
**Pull request by qasmi**
Add cilium installation 
Add Load balancer pool 
Add L2 announcement policy 
Deploy sample app

### CODE REVIEW

**Comment by eddycharly** [comment]
> FYI: the example of LB ip announcement work on linux based OS only

i'm on mac 😓

File: config/cni/cilium/lb-ipam.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: "cilium.io/v2alpha1"
+kind: CiliumLoadBalancerIPPool
+metadata:
+  name: "lb-pool-1"
+spec:
+  cidrs:
+    - cidr: "172.18.250.0/24"
```
Comment (eddycharly) [inline config/cni/cilium/lb-ipam.yaml:7]: @qasmi how did you choose the cidr ?

Comment (qasmi) [reply config/cni/cilium/lb-ipam.yaml:7]: To facilitate routing between the Docker layer and our laptop, it’s recommended to use a subnet from the CIDR used and routed by Docker (default 172.18.0.0/16)
If you run `docker inspect kind` you will see this in the bridge config:
Screenshot 2024-04-25 at 18 08 36

File: play.sh [hunk 2]
```diff
@@ -48,26 +51,65 @@ sure(){
...
+
+  local config=$(gum file "$APP")
+  kubectl apply -f $config
+  checkService
+}
+
+checkService(){
+  SERVICES=$(kubectl get svc -o=jsonpath='{range .items[?(@.spec.type=="LoadBalancer")]}{.metadata.name}{"\n"}{end}')
+  SVC=$(gum choose --header "Which service you want to check ?"  $SERVICES)
+  SVC_IP=$(kubectl get svc sampleservice -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
```
Comment (eddycharly) [inline play.sh]: Service name should not be hardcoded here

**Pull request merged**